modules/org.restlet.ext.apispark/src/org/restlet/ext/apispark/internal/conversion/swagger/v1_2/SwaggerTranslator.java

```diff
@@ -586,20 +587,26 @@ private static void fillContract(Contract contract,
     private static void fillMainAttributes(Definition definition,
             ResourceListing listing, String basePath) {
         definition.setVersion(listing.getApiVersion());
-        Contact contact = new Contact();
-        contact.setEmail(listing.getInfo().getContact());
-        definition.setContact(contact);
-        License license = new License();
-        license.setUrl(listing.getInfo().getLicenseUrl());
-        definition.setLicense(license);
 
         Contract contract = new Contract();
-        contract.setName(listing.getInfo().getTitle());
+        if (listing.getInfo() != null) {
+            Contact contact = new Contact();
+            contact.setEmail(listing.getInfo().getContact());
+            definition.setContact(contact);
+
+            License license = new License();
+            license.setUrl(listing.getInfo().getLicenseUrl());
+            definition.setLicense(license);
+
+            contract.setName(listing.getInfo().getTitle());
+            contract.setDescription(listing.getInfo().getDescription());
+        }
+
         LOGGER.log(Level.FINE, "Contract " + contract.getName() + " added.");
-        contract.setDescription(listing.getInfo().getDescription());
         definition.setContract(contract);
 
-        if (definition.getEndpoints().isEmpty()) {
+        if (definition.getEndpoints().isEmpty()
+                && listing.getAuthorizations() != null) {
```

Comment:
why not check listing.getAuthorizations() != null after having create the endpoint (does it make sens to have an endpoint without authentication info? for me yes)